ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py

```diff
@@ -123,6 +128,25 @@ def get_anta_results(
     )
 
 
+def load_custom_catalogs(catalog_files: list[Path]) -> dict:
+    """Load the custom ANTA catalogs from the provided files and merge them together.
+
+    Args:
+    ----
+      catalog_files (list[Path]): List of Path objects pointing to the custom ANTA catalog files.
+
+    Returns:
+    -------
+      custom_catalog (dict): The merged custom ANTA catalog.
+    """
+    custom_catalog = {}
+    for file in catalog_files:
+        with file.open("r", encoding="UTF-8") as fd:
+            data = load(fd, Loader=CSafeLoader)
+        custom_catalog = merge_catalogs(custom_catalog, data)
```

Comment:
Instead of merging the catalogs one by one, can we load all the data first and then merge them all at once

Reply:
Fixed in 0a009d3d3df3f4c0a577950c59c33d2d308d04b3